cmd/charmd/main.go

```diff
@@ -138,6 +114,17 @@ func serve(confPath string) error {
 	return http.ListenAndServe(conf.APIAddr, debug.Handler("", server))
 }
 
+func addPublicKey(ring *bakery.PublicKeyRing, loc string, key *bakery.PublicKey) error {
+	if key != nil {
+		return ring.AddPublicKeyForLocation(loc, false, key)
+	}
+	pubKey, err := httpbakery.PublicKeyForLocation(http.DefaultClient, loc)
```

Comment:
Since you're using httpbakery.NewPublicKeyRing above, you could just delete lines 121 to 125. Depends how much it matters to get an early failure when the charmstore starts up with no public keys available. Perhaps better to start up OK then fail later, but arguable.